aws/resource_aws_glue_dev_endpoint_test.go

```diff
@@ -0,0 +1,968 @@
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSGlueDevEndpointDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccGlueDevEndpointConfig_Basic(rName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSGlueDevEndpointExists(resourceName, &endpoint),
+					testAccCheckResourceDisappears(testAccProvider, resourceAwsGlueDevEndpoint(), resourceName),
+				),
+				ExpectNonEmptyPlan: true,
+			},
+		},
+	})
+}
+
+func testAccCheckAWSGlueDevEndpointExists(resourceName string, endpoint *glue.DevEndpoint) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("not found: %s", resourceName)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("no ID is set")
+		}
+
+		conn := testAccProvider.Meta().(*AWSClient).glueconn
+		output, err := conn.GetDevEndpoint(&glue.GetDevEndpointInput{
+			EndpointName: aws.String(rs.Primary.ID),
+		})
+
+		if err != nil {
+			return err
+		}
+
+		if output == nil {
+			return fmt.Errorf("no Glue Dev Endpoint")
+		}
+
+		*endpoint = *output.DevEndpoint
+
+		return nil
+	}
+}
+
+func testAccCheckAWSGlueDevEndpointDestroy(s *terraform.State) error {
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_glue_dev_endpoint" {
+			continue
+		}
+
+		conn := testAccProvider.Meta().(*AWSClient).glueconn
+		output, err := conn.GetDevEndpoint(&glue.GetDevEndpointInput{
+			EndpointName: aws.String(rs.Primary.ID),
+		})
+
+		if err != nil {
+			if isAWSErr(err, glue.ErrCodeEntityNotFoundException, "") {
+				return nil
+			}
+			return err
+		}
+
+		endpoint := output.DevEndpoint
+		if endpoint != nil && aws.StringValue(endpoint.EndpointName) == rs.Primary.ID {
+			return fmt.Errorf("the Glue Dev Endpoint %s still exists", rs.Primary.ID)
+		}
+
+		return nil
+	}
+
+	return nil
+}
+
+func testAccGlueDevEndpointConfig_Base(rName string) string {
+	return fmt.Sprintf(`
+resource "aws_iam_role" "test" {
+  name               = "AWSGlueServiceRole-%s"
+  assume_role_policy = data.aws_iam_policy_document.service.json
+}
+
+data "aws_iam_policy_document" "service" {
+  statement {
+    actions = ["sts:AssumeRole"]
+
+    principals {
+      type        = "Service"
+      identifiers = ["glue.amazonaws.com"]
+    }
+  }
+}
+
+resource "aws_iam_policy" "test" {
+  name   = %q
+  policy = data.aws_iam_policy_document.test.json
+}
+
+data "aws_iam_policy_document" "test" {
+  statement {
+    actions   = ["ec2:DescribeSecurityGroups", "ec2:DescribeSubnets"]
+    resources = ["*"]
+  }
+}
+
+resource "aws_iam_role_policy_attachment" "test" {
+  role       = aws_iam_role.test.name
+  policy_arn = aws_iam_policy.test.arn
+}
+
+resource "aws_iam_role_policy_attachment" "glue_service_role" {
+  policy_arn = "arn:aws:iam::aws:policy/service-role/AWSGlueServiceRole"
```

Comment:
lets add

`data "aws_partition" "current" {}`

and use `arn:${data.aws_partition.current.partition}:iam::aws:policy/service-role/AWSGlueServiceRole` so test could run in other partitions as well.